### DISCUSSION TITLE
Jordan Harband - ljharb, npm ecosystem

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#323] [Announcements]
- https://www.linkedin.com/in/ljharb/
- @ljharb
- https://x.com/ljharb

## Questions 

- [x] You maintain a mountain of packages in the npm ecosystem. I think i've seen your profile pic more than any other (maybe minus Sindre). What was the journey that got you to be one of the maintainers of the ecosystem?

### TC39

- [x] You're on the TC39. How did you become a part of that organization?
- [x] What's your area of specialization?
- [x] What contribution are you proudest of?

## Legacy Support

- [x] You now famously hold the position of "we should support old node versions for as long as possible". Why do you think that?
- [x] You maintain 6% of npm downloads. How do you decide if you want to take over maintenance of a package? What is the process like?
- [x] Why are breaking changes [harmful to the ecosystem](https://github.com/import-js/eslint-plugin-import/pull/2447#issuecomment-1119864501)? How can we every move forward if we don't break anything?
- [x] A lot of your lower level packages that contribute to the large graphs center around security and locking down prototype methods. Can you explain to us your thinking on the subject?
- [x] What do you think of the recent movement to create "modern" versions of the packages you ship? I've seen things like neotraverse, neoqs, etc.
- [x] What do you have to say about the wild conspiracy theories for why you do this? Are you getting rich by creating packages in the npm ecosystem (\s)? When is 0 day?
- [x] How do you actually manage so many different major release lines for all of these packages? Any tools that help?
- [x] What are your view on automated CVEs? How has this burdened the ecosystem?
- [x] In some people's eyes you both push JS forward with your TC39 work and hold it back via this legacy support. How do you view it?

### Future

- [x] What do you think the future of npm is?

**Comment by hipstersmoothie** [comment]
@ljharb I took a first pass at this. Feel free to suggest edits/additions!

**Reply by ljharb** [reply]
Looks great, very thorough.
